ops/framework.py

```diff
@@ -578,6 +583,10 @@ def __init__(self, storage: Union[SQLiteStorage, JujuStorage],
         else:
             self.charm_dir = pathlib.Path(charm_dir)
 
+        if not isinstance(event_name, str):
+            raise TypeError(f'event_name should be of type str(), not {type(event_name)}')
```

Comment:
which reminds me, we probably should be passing event_name to Framework in `main()`
```
    framework = ops.framework.Framework(store, charm_dir, meta, model, dispatcher.event_name)
```

Reply:
Arguably, it's good that we surface something if `event_name` is None, because that might save the user trouble later if what they're testing is, say, custom event logging. The prometheus test is doing:
![image](https://user-images.githubusercontent.com/6230162/217203132-d2fec439-1f98-4f51-b2d5-3f3c30d27ef9.png)
I propose we pop a warning instead of raising an error, but if someone is instantiating Framework, they should be notified that there's a new argument they should be passing.

Reply:
downside of that is that we'd need to add back all the `foo`'s or ignore the warning in a lot of tests.